receiver/k8sclusterreceiver/internal/collection/nodes.go

```diff
@@ -62,25 +63,62 @@ func getMetricsForNode(node *corev1.Node, nodeConditionTypesToReport, allocatabl
 		})
 	}
 	// Adding 'node allocatable type' metrics
-	for _, nodeAllocatableTypeValue := range allocatableTypesToReport {
-		nodeAllocatableMetric := getNodeAllocatableMetric(nodeAllocatableTypeValue)
-		v1NodeAllocatableTypeValue := corev1.ResourceName(nodeAllocatableTypeValue)
-		metricValue, err := nodeAllocatableValue(node, v1NodeAllocatableTypeValue)
-
-		// metrics will be skipped if metric not present in node or value is not convertable to int64
-		if err != nil {
-			logger.Debug(err.Error())
-		} else {
-			metrics = append(metrics, &metricspb.Metric{
-				MetricDescriptor: &metricspb.MetricDescriptor{
-					Name:        nodeAllocatableMetric,
-					Description: allocatableDesciption[v1NodeAllocatableTypeValue.String()],
-					Type:        metricspb.MetricDescriptor_GAUGE_INT64,
-				},
-				Timeseries: []*metricspb.TimeSeries{
-					utils.GetInt64TimeSeries(metricValue),
-				},
-			})
+	if featuregate.IsEnabled(reportCPUMetricsAsDoubleFeatureGateID) {
```

Comment:
Please apply similar approach here to reduce duplicated code

Reply:
Done. This one wasn't as clean, but it is more DRY.